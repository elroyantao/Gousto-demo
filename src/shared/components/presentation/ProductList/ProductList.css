.ProductList {
  padding: 0;
  margin: 20px 0 0;
}
