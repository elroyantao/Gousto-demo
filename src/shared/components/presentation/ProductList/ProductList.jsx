import React, { Component } from 'react'
import PropTypes from 'prop-types'

import Product from '../Product/Product'

export default class ProductList extends Component {
  static propTypes = {
    products: PropTypes.array
  }

  static defaultProps = {
    products: []
  }

  render() {
    const { products } = this.props
    return (
      <ul className="ProductList">
        {products.map((product) => <Product key={product.id} product={product} />)}
      </ul>
    )
  }
}
