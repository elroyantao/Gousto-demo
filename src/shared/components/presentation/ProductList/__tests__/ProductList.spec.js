import testComponentHelper from '../../../../lib/test-helper'
import ProductList from '../ProductList'

describe('<ProductList/>', () => {
  const renderComponent = testComponentHelper(ProductList)
  const initialProps = {
    products: [
      { id: '1' },
      { id: '2' },
      { id: '3' },
      { id: '4' }
    ]
  }
  describe('@renders', () => {
    it('in default state', () => {
      expect(renderComponent(initialProps).getTree())
        .toMatchSnapshot()
    })
    it('with empty products', () => {
      expect(renderComponent({}).getTree())
        .toMatchSnapshot()
    })
  })
})
