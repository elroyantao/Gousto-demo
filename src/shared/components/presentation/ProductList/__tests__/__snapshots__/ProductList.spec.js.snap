// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductList/> @renders in default state 1`] = `
<ul
  className="ProductList"
>
  <Product
    product={
      Object {
        "id": "1",
      }
    }
  />
  <Product
    product={
      Object {
        "id": "2",
      }
    }
  />
  <Product
    product={
      Object {
        "id": "3",
      }
    }
  />
  <Product
    product={
      Object {
        "id": "4",
      }
    }
  />
</ul>
`;

exports[`<ProductList/> @renders with empty products 1`] = `
<ul
  className="ProductList"
/>
`;
